src/util/string.h

```diff
@@ -251,11 +252,11 @@ inline std::string trim(const std::string &str)
 {
 	size_t front = 0;
 
-	while (std::isspace(str[front]))
+	while (std::iswspace(str[front]))
```

Comment:
It stops minetest from being crashed by unicode in the configuration files, but you are right that this isn't really the correct way to go about fixing that.